### PR TITLE
Feature/bounty

### PR DESCRIPTION
**Pull request by Velenir**
Config and migrations for deployment on mainnet for bug bounty

### CODE REVIEW

File: migrations/11_SCHEME-REP-Lock_Whitelisted_tokens.js [hunk 1]
```diff
@@ -88,18 +88,30 @@ module.exports = async function (deployer, network) {
...
+        OMG: '0x00df91984582e6e96288307e9c2f20b38c8fece9',
+        testGEN: '0xa1f34744c80e7a9019a5cd2bf697f13df00f9773',
+        GEN: '0x543ff227f64aa17ea132bf9886cab5db55dcaddf'
+      }
+      // to allow for dry-runs on mainnet-fork
+    } else if (network.startsWith('mainnet')) {
+      whitelistedTokens = {
+        GNO: '0x6810e776880C02933D47DB1b9fc05908e5386b96',
+        GEN: '0x543ff227f64aa17ea132bf9886cab5db55dcaddf'
+      }
```
Comment (Velenir) [inline migrations/11_SCHEME-REP-Lock_Whitelisted_tokens.js]: Mainnet GEN and GNO to whitelist

File: migrations/12_SCHEMES_GenericScheme.js [hunk 2]
```diff
@@ -20,15 +20,22 @@ module.exports = async function (deployer) { // eslint-disable-line no-unused-va
...
+    address: genesisProtocolAddress
+  } = await genesisProtocolHelper.getGenesisProtocolData('dutchX')
 
   // DutchX address
-  const dutchXContractAddress = await getDXContractAddress('DutchExchangeProxy')
+  let dutchXContractAddress
+  if (process.env.USE_MOCK_DX) {
+    const Wallet = artifacts.require('Wallet')
+    const WalletDeployed = await Wallet.deployed()
+    dutchXContractAddress = WalletDeployed.address
```
Comment (Velenir) [inline migrations/12_SCHEMES_GenericScheme.js]: Deploying Wallet as Mock DX

File: migrations/15_Configure_Oracle_and_set_Avatar_as_DX_auctioneer.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+      const address = whitelistedTokens[tokenName]
+      console.log('  - %s: %s', tokenName, address, `to 1/${i+1}`)
+
+      tokens.push(address)
+      nums.push(1)
+      dens.push(i + 1)
+    })
+
+    if (tokens.length > 0) {
+      await PriceOracle.setPrices(tokens, nums, dens)
```
Comment (Velenir) [inline migrations/15_Configure_Oracle_and_set_Avatar_as_DX_auctioneer.js]: Setting price as 1/1 for GNO and 1/2 for GEN

File: migrations/15_Configure_Oracle_and_set_Avatar_as_DX_auctioneer.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+
+  if (process.env.USE_MOCK_DX) {
+    const Wallet = artifacts.require('Wallet')
+    const WalletDeployed = await Wallet.deployed()
+
+    const DxAvatar = artifacts.require('DxAvatar')
+    const dxAvatar = await DxAvatar.deployed()
+
+    console.log('Setting Avatar at address', dxAvatar.address, 'as Wallet\'s owner');
+    await WalletDeployed.transferOwnership(dxAvatar.address)
```
Comment (Velenir) [inline migrations/15_Configure_Oracle_and_set_Avatar_as_DX_auctioneer.js]: Make Avatar into Wallet's owner

File: migrations/2_DEV_local_development.js [hunk 5]
```diff
@@ -44,6 +44,10 @@ module.exports = async function (deployer, network, accounts) {
 
     // Deploy GenesisProtocol
     await deployGenesisProtocol(deployer)
+  } else if (process.env.USE_MOCK_DX) {
+    console.log('Deploying Wallet as Mock DutchX');
+    const Wallet = artifacts.require('Wallet')
+    await deployer.deploy(Wallet)
```
Comment (Velenir) [inline migrations/2_DEV_local_development.js]: Deploying Wallet as Mock DX

File: env_vars/bounty.31.01.2019.env [hunk 6]
```diff
@@ -0,0 +1,75 @@
+# Allow to use native solc docker image
```
Comment (Velenir) [inline env_vars/bounty.31.01.2019.env]: New env vars

File: migrations/12_SCHEMES_GenericScheme.js (hunk 2 above)
Comment (orenyodfat) [reply migrations/12_SCHEMES_GenericScheme.js]: do you have a separate script to transfer ownership to the dao ?

File: migrations/15_Configure_Oracle_and_set_Avatar_as_DX_auctioneer.js (hunk 3 above)
Comment (orenyodfat) [reply migrations/15_Configure_Oracle_and_set_Avatar_as_DX_auctioneer.js]: Who is the owner of the FixPriceOracle ? I thought the plan is to use daostack priceOracleMock  , which will be owned by a multisg

Comment (Velenir) [reply migrations/15_Configure_Oracle_and_set_Avatar_as_DX_auctioneer.js]: Deploying account is the owner.

The plan was to use fix price oracle, unless it was changed afterwards. The bounty doc specifically mentions a fix oracle, same as was used before.

File: migrations/12_SCHEMES_GenericScheme.js (hunk 2 above)
Comment (Velenir) [reply migrations/12_SCHEMES_GenericScheme.js]: Ownership is transferred in migration `15`, ignore that it was commented out. Bugged out in dry-run, that's all

File: package.json [hunk 7]
```diff
@@ -49,7 +49,7 @@
   "dependencies": {
     "@daostack/arc": "0.0.1-rc.8",
     "@daostack/migration": "0.0.1-rc.8-v2",
-    "@gnosis.pm/dx-contracts": "^2.0.0-RC15",
+    "@gnosis.pm/dx-contracts": "gnosis/dx-contracts#workaround/2.0_PriceOracleInterface_name_collision",
```
Comment (Velenir) [inline package.json]: temp workaround
better remove when either dx-contracts or arc changes PriceOracleInterface name

**Comment by Velenir** [comment]
Not relevant anymore

**Pull request closed without merging**